docs/concepts/services-networking/service.md

```diff
@@ -203,6 +203,35 @@ having working [readiness probes](/docs/tasks/configure-pod-container/configure-
 
 ![Services overview diagram for iptables proxy](/images/docs/services-iptables-overview.svg)
 
+### Proxy-mode: ipvs
+
+In this mode, kube-proxy watches Kubernetes `services` and `endpoints`,
+call `netlink` interface create ipvs `virtual server` and `real server` accordingly.
+Kube-proxy will also sync local virtual server and real server with Kubernetes
+`services` and `endpoints`  periodically, to make sure ipvs status is
+consistent with the expectation. When access the `service`, traffic will
+be redirect to one of the backend `pod`.
+
+Ipvs based on netfilter hook function, like iptables proxy, but use hash
+table as the underlying data structure and work in the kernal state.
+That means ipvs redirects traffic can be much faster, and have much
+better performance when sync proxy rules. Furthermore, ipvs provides more
+options for load balancing algorithm, such as:
+
+- rr: round-robin
+- lc: least connection
+- dh: destination hashing
+- sh: source hashing
+- sed: shortest expected delay
+- nq: never queue
+
+**Note:** ipvs mode assumed IPVS kernel modules are installed on the node
+before running kube-proxy. When kube-proxy starts, if proxy mode is ipvs,
+kube-proxy would validate if IPVS modules are installed on the node, if
+it's not installed kube-proxy will fall back to userspace proxy mode.
```

Comment:
userspace -> iptables

Reply:
ipvs.CanUseIPVSProxier will return (false, error) if ipvs modules are not installed. so kube-proxy will fall back to userspace proxy mode. 

I am not sure if I am missing something. @m1093782566 @dujun1990 

```
func tryIPVSProxy(iptver iptables.IPTablesVersioner, kcompat iptables.KernelCompatTester) string {
	// guaranteed false on error, error only necessary for debugging
	// IPVS Proxier relies on iptables
	useIPVSProxy, err := ipvs.CanUseIPVSProxier()
	if err != nil {
		utilruntime.HandleError(fmt.Errorf("can't determine whether to use ipvs proxy, using userspace proxier: %v", err))
		return proxyModeUserspace
	}
	if useIPVSProxy {
		return proxyModeIPVS
	}

	// TODO: Check ipvs version

	// Try to fallback to iptables before falling back to userspace
	glog.V(1).Infof("Can't use ipvs proxier, trying iptables proxier")
	return tryIPTablesProxy(iptver, kcompat)
}
```

Reply:
Sorry, that's not intended - I assume it's a bug. See bug fix in #54120. FYI